### PR TITLE
Add desktop content switcher to the home page

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Related to #592
## Description

This PR adds a Content Type switcher to the desktop version of the homepage. I've done it to test out how well the content switcher popover works when not in the header, and to see how best to handle state. It turns out that to make the component work well both in the header and on the homepage, it is easier to handle state (i.e. currently selected item) of the `VContentTypePopover` from the parent.

## Testing Instructions

Run `pnpm run dev`, and open the home page. You should be able to see the content switcher inside the search bar, and also to select a content type from the dropdown (popover). The search itself is not functioning yet.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [ ] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by sarayourfriend — COMMENTED** [review]
How does this PR related to https://github.com/WordPress/openverse-frontend/pull/607 ?

File: src/composables/use-content-type.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+const icons = {
+  all: allIcon,
+  audio: audioIcon,
+  image: imageIcon,
+}
+export default function useContentType() {
+  const { store } = useContext()
+
+  const contentTypes = reactive(['all', 'image', 'audio'])
```
[the remark this answers is not in the document]
Comment (obulat) [reply src/composables/use-content-type.js:16]: True, this should be a static array, in fact, there should be a single source of truth for the content type list. I will change this.

**Comment by obulat** [comment]
> How does this PR related to #607 ?

I have been thinking about the Content Switcher only as part of the Header for a very long time. And just a couple of days ago I looked at the homepage, and realized that the same component is used inside the search bar. This made me rethink the organization of all of the components related to the switching of content types.
I will use the changes from this PR in #607.

**Review by zackkrida — APPROVED** [review]
Oh, what an amazing surprise, I assumed this would be on the _old_ homepage! This is great. @dhruvkb I want to make sure you're aware of this since you're working on finishing the homepage.

**Comment by zackkrida** [comment]
@dhruvkb feel free to merge this without additional review if it is helpful to your homepage PR; the specific details can be discussed there.

**Comment by dhruvkb** [comment]
It will definitely be very useful, let me resolve the conflicts and merge it. Tagging @obulat for the notification.

**Pull request merged**